src/core/tiledmesh/qgscesiumutils.cpp

```diff
@@ -45,3 +46,125 @@ QgsBox3d QgsCesiumUtils::parseRegion( const QVariantList &region )
 
   return parseRegion( QgsJsonUtils::jsonFromVariant( region ) );
 }
+
+QgsOrientedBoundingBox QgsCesiumUtils::parseBox( const json &box )
+{
+  if ( box.size() != 12 )
+    return QgsOrientedBoundingBox();
+
+  return QgsOrientedBoundingBox::fromJson( box );
+}
+
+QgsOrientedBoundingBox QgsCesiumUtils::parseBox( const QVariantList &box )
+{
+  if ( box.size() != 12 )
+    return QgsOrientedBoundingBox();
+
+  return QgsOrientedBoundingBox::fromJson( QgsJsonUtils::jsonFromVariant( box ) );
+}
+
+QgsSphere QgsCesiumUtils::parseSphere( const json &sphere )
+{
+  if ( sphere.size() != 4 )
+    return QgsSphere();
+
+  try
+  {
+    const double centerX = sphere[0].get<double>();
+    const double centerY = sphere[1].get<double>();
+    const double centerZ = sphere[2].get<double>();
+    const double radius = sphere[3].get<double>();
+    return QgsSphere( centerX, centerY, centerZ, radius );
+  }
+  catch ( nlohmann::json::exception & )
+  {
+    return QgsSphere();
+  }
+}
+
+QgsSphere QgsCesiumUtils::parseSphere( const QVariantList &sphere )
+{
+  if ( sphere.size() != 4 )
+    return QgsSphere();
+
+  return parseSphere( QgsJsonUtils::jsonFromVariant( sphere ) );
+}
+
+//
+// QgsOrientedBoundingBox
+//
+
+QgsOrientedBoundingBox QgsOrientedBoundingBox::fromJson( const json &json )
+{
+  try
+  {
+    QgsOrientedBoundingBox res;
+    for ( int i = 0; i < 3; ++i )
+    {
+      res.mCenter[i] = json[i].get<double>();
+    }
+    for ( int i = 0; i < 9; ++i )
+    {
+      res.mHalfAxes[i] = json[i + 3].get<double>();
+    }
+    return res;
+  }
+  catch ( nlohmann::json::exception & )
+  {
+    return QgsOrientedBoundingBox();
+  }
+}
+
+QgsOrientedBoundingBox::QgsOrientedBoundingBox() = default;
+
+QgsOrientedBoundingBox::QgsOrientedBoundingBox( const QList<double> &center, QList<double> &halfAxes )
+{
+  if ( center.size() == 3 )
+  {
+    mCenter[0] = center.at( 0 );
+    mCenter[1] = center.at( 1 );
+    mCenter[2] = center.at( 2 );
+  }
+  if ( halfAxes.size() == 9 )
+  {
+    for ( int i = 0; i < 9; ++i )
+    {
+      mHalfAxes[i] = halfAxes.at( i );
+    }
+  }
+}
+
+bool QgsOrientedBoundingBox::isNull() const
+{
+  return std::isnan( mCenter[0] ) || std::isnan( mCenter[1] ) || std::isnan( mCenter[2] );
+}
+
+QList< double > QgsOrientedBoundingBox::halfAxesList() const
```

Comment:
maybe QVector<double> would be better?

Reply:
I added this one for python convenience only -- in C++ we should just use the double array directly.